projects/plugins/boost/app/modules/cache/pre-wordpress/storage/File_Storage.php

```diff
@@ -58,6 +60,24 @@ public function read( $request_uri, $parameters ) {
 		return false;
 	}
 
+	/**
+	 * Garbage collect expired files.
+	 */
+	public function garbage_collect() {
+		$cache_duration = apply_filters( 'jetpack_boost_cache_duration', 3600 );
+
+		if ( $cache_duration === 0 ) {
+			// Garbage collection is disabled.
+			return false;
+		}
+
+		clearstatcache();
```

Comment:
Should this be inside `delete_expired_files`?

I guess you left it out to avoid having to call it each "level" of recursion? Just have a public function `delete_expired_files` in `Filesystem_Utils` which calls a private recursive function that does all the work - but that  way the entry point can clear the statcache on the way in.